src/sentry/relay/config.py

```diff
@@ -345,3 +349,70 @@ def _filter_option_to_config_setting(flt, setting):
                 # ret_val['options'] = setting.split(' ')
                 ret_val["options"] = list(setting)
     return ret_val
+
+
+# Note: names for extracted metrics start with `measurement.`, while the
+# in-event keys start with `measurements.`
+ALL_MEASUREMENT_METRICS = frozenset(
+    [
+        "measurement.fp",
+        "measurement.fcp",
+        "measurement.lcp",
+        "measurement.fid",
+        "measurement.cls",
+        "measurement.ttfb",
+        "measurement.ttfb.requesttime",
+        "measurement.app_start_cold",
+        "measurement.app_start_warm",
+        "measurement.frames_total",
+        "measurement.frames_slow",
+        "measurement.frames_frozen",
+        "measurement.frames_slow_rate",
+        "measurement.frames_frozen_rate",
+        "measurement.stall_count",
+        "measurement.stall_total_time",
+        "measurement.stall_longest_time",
+        "measurement.stall_percentage",
+    ]
+)
+
+
+def get_transaction_metrics_settings(
+    project: Project, breakdowns_config: Optional[Mapping[str, Any]]
+):
+    metrics = []
+    custom_tags = []
+
+    if features.has("organizations:transaction-metrics-extraction", project.organization):
+        # TODO: for now let's extract all known measurements. we might want to
+        # be more fine-grained in the future once we know which measurements we
+        # really need (or how that can be dynamically determined)
+        metrics.extend(ALL_MEASUREMENT_METRICS)
+
+        if breakdowns_config is not None:
+            # we already have a breakdown configuration that tells relay which
+            # breakdowns to compute for an event. metrics extraction should
+            # probably be in sync with that, or at least not extract more metrics
+            # than there are breakdowns configured.
+            try:
+                for breakdown_name, breakdown_config in breakdowns_config.items():
+                    assert breakdown_config["type"] == "spanOperations"
+
+                    for op_name in breakdown_config["matches"]:
+                        metrics.append(f"breakdown.{breakdown_name}.{op_name}")
+            except Exception:
+                capture_exception()
```

Comment:
Would it make more sense to move the `try ... except` inside the loop so if an error occurs, we don't abort on the rest of the items in `breakdowns_config`?

Reply:
if `breakdowns_config` is not a dict we want that error to be caught too. we shouldn't rely on this anyway, this is just so event ingestion isn't disrupted if somebody messes up the config